lib/client.js

```diff
@@ -119,8 +136,11 @@ function wrappedPostMessage (name, params) {
     rawPostMessage(this, msg)
   })
 
-  // ensure promise is cleaned up when resolved
-  return promise.then(removePromise.bind(null, id, timeoutId), removePromise.bind(null, id, timeoutId))
+  // ensure promise is cleaned up when resolved/rejected, pass the start time for resolved promises
+  return promise.then(
+    removePromise.bind(null, id, timeoutId, this, actions, promiseStartTime),
+    removePromise.bind(null, id, timeoutId)
```

Comment:
Question: not related to this PR, but why do we have to do this?

Reply:
Which part? It looks as though previously we were passing the same method (with the same params) as both the `resolve` and `reject` callbacks to this promise, though that probably could have been handled via a `finally` (or equivalent) handler instead. Now we're passing additional params when the promise is successfully resolved, and omitting those params in the rejected handler. How might you rewrite this?

Reply:
Ignore this. I totally missed `promise.then` and thought it was `promose.all` and assumed that these are multiple promises.